src/main/java/me/qoomon/maven/gitversioning/GitVersioningModelProcessor.java

```diff
@@ -338,6 +339,14 @@ private Model processModel(Model projectModel) throws IOException {
             }
 
             this.virtualProjectModelCache.put(projectId, virtualProjectModel);
+
+            for (Model model : virtualProjectModelCache.values()) {
+                if(projectModel != model){
+                    MavenXpp3Writer writer = new MavenXpp3Writer();
+                    File file = new File(model.getPomFile().getParent(), GIT_VERSIONING_POM_NAME);
+                    writer.write(new FileOutputStream(file), model);
+                }
+            }
```

Comment:
@qoomon Why we are not writing pom.file in the first place here? We have model and we can easily persist it. The mojo can only switch to this file. That error prone logic will disappear on the VersioningMojo file.

Reply:
By the way I put near original file :)

Reply:
I like the idea of only switching the pom file later in the mojo and generate it in the processor.
However I don't wanna change pom structure in any way or losing comments. This is because the pom file should stay as untouched as possible.

Reply:
I thought about your reason and I feel you are right. Can you take a look this project? It is not published but for our purpose it seems nice. I checked their test codes and seems good. It is not published. I think you can clone this project and shade it and then republish to maven central to safely depends on.

https://github.com/CoreMedia/maven-jdom-parser